raiden-ts/src/channels/epics.ts

```diff
@@ -607,14 +613,24 @@ const makeDeposit$ = (
   approveNonce?: number,
 ): Observable<channelDeposit.failure> => {
   if (!deposit?.gt(Zero)) return EMPTY;
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  function stopRetryIfRejected(err: any, count: number) {
+    // 4001 is Metamask's code for rejected/denied signature prompt
+    return count >= 10 || err?.code === 4001;
+  }
+
   return defer(() => tokenContract.functions.allowance(sender, tokenNetworkContract.address)).pipe(
     mergeMap((allowance) =>
       allowance.gte(deposit)
         ? of(true)
-        : from(
-            tokenContract.functions.approve(tokenNetworkContract.address, deposit, {
-              nonce: approveNonce,
-            }),
+        : retryAsync$(
+            () =>
+              tokenContract.functions.approve(tokenNetworkContract.address, deposit, {
+                nonce: approveNonce,
+              }),
+            (tokenContract.provider as JsonRpcProvider).pollingInterval,
+            stopRetryIfRejected,
```

Comment:
Nitpick: Don't hesitate to be more explicit. :wink: 

```suggestion
            stopRetryIfTransactionGotRejected,
```

Reply:
No, please, this isn't english, the original variable already had enough verbosity... this would make many function calls with just 2-3 arguments become multi-line statements to keep the 99char line limit.